polygeist/tools/cgeist/Lib/CGExpr.cc

```diff
@@ -146,6 +146,14 @@ ValueCategory MLIRScanner::VisitParenExpr(clang::ParenExpr *Expr) {
   return Visit(Expr->getSubExpr());
 }
 
+template <typename T>
+mlir::MemRefType MLIRScanner::SYCLGetElementType(T SYCLType, unsigned ElemIndex,
+                                                 llvm::ArrayRef<int64_t> Shape,
+                                                 mlir::Attribute MemorySpace) {
+  return mlir::MemRefType::get(Shape, SYCLType.getBody()[ElemIndex],
```

Comment:
The template could be instantiated with any typename `T`. Is possible to limit the types a template can be instantiated with static_assert so that would be a nice thing to add here.